AFQ/viz/utils.py

```diff
@@ -411,6 +434,615 @@ def visualize_tract_profiles(tract_profiles, scalar="dti_fa", min_fa=0.0,
+
+        for subject in self.subjects:
+            fig, axes = self._get_brain_axes('Subject ' + str(subject))
+            for bundle in bundles:
+                ax = axes[positions[bundle][0], positions[bundle][1]]
+                for name in names:
+                    profile = self._get_profile(name, bundle, subject, scalar)
+                    if profile is not None:
+                        ax.plot(profile)
+                ax.set_title(bundle)
+                ax.set_ylim([min_scalar, max_scalar])
+                y_ticks = np.asarray([0.2, 0.4, 0.6]) * max_scalar
+                ax.set_yticks(y_ticks)
+                ax.set_yticklabels(y_ticks)
+                ax.set_xticklabels([])
+
+            fig.legend(names, loc='center')
+            fig.savefig(
+                self._get_fname(
+                    f"tract_profiles/{scalar}",
+                    f"{'_'.join(names)}_sub-{subject}"))
+
+        if not show_plots:
+            plt.ion()
+
+    def contrast_index(self, names=None, scalar="dti_fa",
+                       bundles=list(POSITIONS.keys()), show_plots=False):
+        """
+        Calculate the contrast index for each bundle in two datasets.
+
+        Parameters
+        ----------
+        names : list of strings, optional
+            Names of datasets to plot profiles of.
+            If None, all datasets are used.
+            Should be a total of only two datasets.
+            Default: None
+
+        scalar : string, optional
+            Scalar to use for the contrast index. Default: "dti_fa".
+
+        bundles : list of strings, optional
+            Bundles to correlate. Default: list(POSITIONS.keys())
+
+        show_plots : bool, optional
+            Whether to show plots if in an interactive environment.
+            Default: False
+
+        Returns
+        -------
+        Pandas dataframe of contrast indices
+        with subjects as columns and bundles as rows.
+        """
+        if not show_plots:
+            plt.ioff()
+
+        if names is None:
+            names = list(self.profile_dict.keys())
+        if len(names) != 2:
+            self.logger.error("To calculate the contrast index, "
+                              + "only two dataset names should be given")
+            return None
+
+        contrast_index = pd.DataFrame(index=bundles, columns=self.subjects)
+        for subject in self.subjects:
+            fig, axes = self._get_brain_axes(
+                (f"Contrast Indices by Bundle, "
+                    f" {names[0]} vs {names[1]}"))
+            for bundle in bundles:
+                profiles = [None] * 2
+                both_found = True
+                for i, name in enumerate(names):
+                    profiles[i] = self._get_profile(
+                        name, bundle, subject, scalar)
+                    if profiles[i] is None:
+                        both_found = False
+                if both_found:
+                    this_contrast_index = \
```

Comment:
Maybe we should make the calculation of a contrast index a utility function that can be reused elsewhere? Maybe put it in the utils module, rather than in here?

Reply:
Where does the contrast index come from? I am thinking some function like this should already exist in numpy or scipy, but I cannot find much information about it online. Is there another name for it?